tests/unit/workflows/test_workflows_tasks_submission.py

```diff
@@ -35,9 +41,51 @@ class DummyEng(object):
     pass
 
 
+def fake_is_arxiv_paper(obj):
+    return True
+
+
 def test_add_note_entry_does_not_add_value_that_is_already_present():
     obj = StubObj({'public_notes': [{'value': '*Temporary entry*'}]}, {'core': 'something'})
     eng = DummyEng()
 
     assert add_note_entry(obj, eng) is None
     assert {'public_notes': [{'value': '*Temporary entry*'}]} == obj.data
+
+
+@mock.patch('inspirehep.modules.workflows.tasks.submission.is_arxiv_paper',
+    side_effect=fake_is_arxiv_paper)
+def test_prepare_file(fake_is_arxiv_paper):
+    class MockFilesCollectionIterator(object):
+
+        def __init__(self, data):
+            self.data = data
+
+        @property
+        def keys(self):
+            return self.data.keys()
+
+        def __getitem__(self, item):
+            return self.data[item]
+
+    eng = DummyEng()
+    files_collection = MagicMock()
+    mock_file = MagicMock()
+
+    mock_file.name = MagicMock(return_value='fake_record')
+    mock_file.obj.file.uri = 'fake_uri'
```

Comment:
I understand what you are doing with the mocks here, but I wonder if these could be pulled up in the `MockObj` definition rather than defined per test.